connection/types.go

```diff
@@ -28,32 +27,34 @@ const (
 
 // probably needs a better name since its not only used for piping the connection
 type PipeConnection interface {
+	getConn() ByteConnection
```

Comment:
Instead of getConn() you could just leave it as conn(). The Go standard library also follows this convention.

Reply:
im aware of the convention but i keep forgetting about it  ':)